### DISCUSSION TITLE
Is there a way to delete a trailing unnamed column?

### DISCUSSION THREAD

**Discussion by davidpellerin** [mithrandie/csvq#66] [General]
Hi 👋 Just wondering if there is a way to delete a blank column caused by a trailing comma in a CSV on each line?

I figure I could use `sed` and give it a name, but i'd rather not modify a vendor supplied file if I don't have to.

**Comment by ondohotola** [comment]
ALTER TABLE, perhaps?

but why bother? Just don’t SELECT it.

el

—

**Comment by kpym** [comment]
You can use `t.1` to access the 1st column of the table `t`. So you can use `alter table` to drop the 1st column.

For example if `test.csv` is the following file
```
> cat test.csv
,A,B,C
,1,2,3
,4,5,6
```
then you can for example drop the first column like this:
```
> cat test.csv | csvq -f csv -q  "alter table stdin drop stdin.1; select * from stdin;"
A,B,C
1,2,3
4,5,6
```
